src/natcap/invest/carbon.py

```diff
@@ -176,6 +197,73 @@
+        },
+        "tot_c_cur.tif": {
+            "about": "Raster showing the amount of carbon stored in each pixel for the current scenario. It is a sum of all of the carbon pools provided by the biophysical table.",
+            "bands": {1: {
+                "type": "number",
+                "units": u.metric_ton/u.pixel
+            }}
+        },
+        "tot_c_fut.tif": {
+            "about": "Raster showing the amount of carbon stored in each pixel for the future scenario. It is a sum of all of the carbon pools provided by the biophysical table.",
+            "bands": {1: {
+                "type": "number",
+                "units": u.metric_ton/u.pixel
+            }},
+            "created_if": "lulc_fut_path"
+        },
+        "tot_c_redd.tif": {
+            "about": "Raster showing the amount of carbon stored in each pixel for the REDD scenario. It is a sum of all of the carbon pools provided by the biophysical table.",
+            "bands": {1: {
+                "type": "number",
+                "units": u.metric_ton/u.pixel
+            }},
+            "created_if": "lulc_redd_path"
+        },
+        "delta_cur_fut.tif": {
+            "about": "Raster showing the difference in carbon stored between the future landscape and the current landscape. In this map some values may be negative and some positive. Positive values indicate sequestered carbon, negative values indicate carbon that was lost.",
+            "bands": {1: {
+                "type": "number",
+                "units": u.metric_ton/u.pixel
+            }},
+            "created_if": "lulc_fut_path"
+        },
+        "delta_cur_redd.tif": {
+            "about": "Raster showing the difference in carbon stored between the REDD landscape and the current landscape. In this map some values may be negative and some positive. Positive values indicate sequestered carbon, negative values indicate carbon that was lost.",
+            "bands": {1: {
+                "type": "number",
+                "units": u.metric_ton/u.pixel
+            }},
+            "created_if": "lulc_redd_path"
+        },
+        "npv_fut.tif": {
+            "about": "Rasters showing the economic value of carbon sequestered between the current and the future landscape dates.",
+            "bands": {1: {
+                "type": "number",
+                "units": u.currency/u.pixel
+            }},
+            "created_if": "lulc_fut_path"
+        },
+        "npv_redd.tif": {
+            "about": "Rasters showing the economic value of carbon sequestered between the current and the REDD landscape dates.",
+            "bands": {1: {
+                "type": "number",
+                "units": u.currency/u.pixel
+            }},
+            "created_if": "lulc_redd_path"
+        },
+        "intermediate": {
+            "type": "directory",

```

Comment:
In `test_model_specs` we assume it's a file unless `"type"` is set to `"directory"`.
- If the `type` key exists, use that
- Otherwise, infer the type from the file extension
- If there's no valid file extension, raise an error
https://github.com/emlys/invest/blob/7a2fc54d07562d5ad979e0f84f64429bc2a724a2/tests/test_model_specs.py#L103-L119